// reducers/categories.js
const GET_CATEGORIES_START = 'GET_CATEGORIES_START'
const GET_CATEGORIES_SUCCESS = 'GET_CATEGORIES_SUCCESS'
const GET_CATEGORIES_FAILURE = 'GET_CATEGORIES_FAILURE'

import {getCategories as getCategoriesApi} from 'api'

let initialState = {
    categories: [],
    loading: false
}

export const categories = (state = initialState, { type, payload }) => {
    switch (type) {
        case GET_CATEGORIES_START:
            return {
                ...state,
                loading: true
            }
        case GET_CATEGORIES_SUCCESS:
            return {
                ...state,
                loading: false,
                categories: payload
            }
        default:
            return state
    }
}

export const getCategories = () => async dispatch => {
    dispatch({
        type: GET_CATEGORIES_START
    })
    try {
        const categories = await getCategoriesApi()
        dispatch({
            type: GET_CATEGORIES_SUCCESS,
            payload: categories
        })
    } catch (err) {
        dispatch({
            type: GET_CATEGORIES_FAILURE,
            payload: err,
            error: true
        })
    }
}
