// reducers/meal.js
const GET_MEAL_BY_ID_START = 'GET_MEAL_BY_ID_START'
const GET_MEAL_BY_ID_SUCCESS = 'GET_MEAL_BY_ID_SUCCESS'
const GET_MEAL_BY_ID_FAILURE = 'GET_MEAL_BY_ID_FAILURE'

import { getMealById as getMealByIdApi } from 'api'

let initialState = {
    meal: [],
    loading: false
}

export const meal = (state = initialState, { type, payload }) => {
    switch (type) {
        case GET_MEAL_BY_ID_START:
            return {
                ...state,
                loading: true
            }
        case GET_MEAL_BY_ID_SUCCESS:
            return {
                ...state,
                meal: payload,
                loading: false
            }
        default:
            return state
    }
}

export const getMealById = id => async dispatch => {
    dispatch({
        type: GET_MEAL_BY_ID_START
    })
    try {
        const meal = await getMealByIdApi(id)
        dispatch({
            type: GET_MEAL_BY_ID_SUCCESS,
            payload: meal
        })
    } catch (err) {
        dispatch({
            type: GET_MEAL_BY_ID_FAILURE,
            payload: err,
            error: true
        })
    }
}
