// reducers/meals.js
const GET_MEALS_START = 'GET_MEALS_START'
const GET_MEALS_SUCCESS = 'GET_MEALS_SUCCESS'
const GET_MEALS_FAILURE = 'GET_MEALS_FAILURE'

import { 
    getMealsByCategory as getMealsByCategoryApi,
    getMealsByName as getMealsByNameApi
} from 'api'

let initialState = {
    meals: [],
    loading: false
}

export const meals = (state = initialState, { type, payload }) => {
    switch (type) {
        case GET_MEALS_START:
            return {
                ...state,
                loading: true
            }
        case GET_MEALS_SUCCESS:
            return {
                ...state,
                meals: payload,
                loading: false
            }
        default:
            return state
    }
}

export const getMealsByCategory = category => async dispatch => {
    dispatch({
        type: GET_MEALS_START
    })
    try {
        const meals = await getMealsByCategoryApi(category)
        dispatch({
            type: GET_MEALS_SUCCESS,
            payload: meals
        })
    } catch (err) {
        dispatch({
            type: GET_MEALS_FAILURE,
            payload: err,
            error: true
        })
    }
}

export const getMealsByName = name => async dispatch => {
    dispatch({
        type: GET_MEALS_START
    })
    try {
        const meals = await getMealsByNameApi(name)
        dispatch({
            type: GET_MEALS_SUCCESS,
            payload: meals
        })
    } catch (err) {
        dispatch({
            type: GET_MEALS_FAILURE,
            payload: err,
            error: true
        })
    }
}
