// reducers/recipe.js
const GET_RECIPE_BY_ID_START = 'GET_RECIPE_BY_ID_START'
const GET_RECIPE_BY_ID_SUCCESS = 'GET_RECIPE_BY_ID_SUCCESS'
const GET_RECIPE_BY_ID_FAILURE = 'GET_RECIPE_BY_ID_FAILURE'

import { getRecipeById as getRecipeByIdApi } from 'api'

let initialState = {
    recipe: [],
    loading: false
}

export const recipe = (state = initialState, { type, payload }) => {
    switch (type) {
        case GET_RECIPE_BY_ID_START:
            return {
                ...state,
                loading: true
            }
        case GET_RECIPE_BY_ID_SUCCESS:
            return {
                ...state,
                recipe: payload,
                loading: false
            }
        default:
            return state
    }
}

export const getRecipeById = id => async dispatch => {
    dispatch({
        type: GET_RECIPE_BY_ID_START
    })
    try {
        const recipe = await getRecipeByIdApi(id)
        dispatch({
            type: GET_RECIPE_BY_ID_SUCCESS,
            payload: recipe
        })
    } catch (err) {
        dispatch({
            type: GET_RECIPE_BY_ID_FAILURE,
            payload: err,
            error: true
        })
    }
}
