// components/RecipePage/Recipe/index.jsx
import React from 'react'

function getIngredients (props) {
    const ingredients = {}
    for (let key in props) {
        if (key.includes('strIngredient') && props[key] !== '') {
            ingredients[props[key]] = 0
        }
        if (key.includes('strMeasure') && props[key] !== '') {
            for (let elem in ingredients) {
                if (ingredients[elem] === 0) {
                    ingredients[elem] = props[key]
                    break
                }
            }
        }
    }
    return ingredients
}

function Recipe(props) {
    const {
        idMeal: id,
        strArea: area,
        strCategory: category,
        strInstructions: instr,
        strMeal: name,
        strMealThumb: image,
        strYoutube: video
    } = props
    let ingredients = getIngredients(props)

    return (
        <div className='recipe'>
            <img style={{width: '400px'}} src={image} alt={name} />
            <h1>{name}</h1>
            <h6>Category: {category}</h6>
            {area ? <h6>Area: {area}</h6> : ''}
            <p>{instr}</p>
            <table className='left'>
                <thead>
                    <tr>
                        <th>Ingredient</th>
                        <th>Measures</th>
                    </tr>
                </thead>
                <tbody>
                    {
                        Object.keys(ingredients).map(key => {
                            return <tr key={key}>
                                <td>{key}</td>
                                <td>{ingredients[key]}</td>
                            </tr>
                        })
                    }
                </tbody>
            </table>
            {video ? (
                <div className="row">
                    <h5 style={{margin: '2rem 0 1.5rem'}}>Video Recipe</h5>
                    <iframe title={id} src={`https://www.youtube.com/embed/${video.slice(-11)}`} allowFullScreen/>
                </div>
            ) : ''}
        </div>
    )
}

export default (Recipe)
